src/cancellation_manager.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+
+#ifndef MEMORY_AGENT_CANCELLATION_MANAGER_H
+#define MEMORY_AGENT_CANCELLATION_MANAGER_H
+
+#include <chrono>
+#include <string>
+
+class CancellationManager {
+public:
+    CancellationManager(const std::string &, const std::chrono::steady_clock::time_point &);
```

Comment:
Since it is not prohibited to specify parameters names we can specify them to understand the semantics much easier